code/game/objects/structures/teleporter.dm

```diff
@@ -160,6 +184,26 @@
 		CRASH("A teleporter was linked with itself!")
 	src.linked_teleporter = linked_teleporter
 
+/obj/item/teleporter_kit/attackby(obj/item/I, mob/user, params)
+	if(!ishuman(user))
+		return FALSE
+	if(!istype(I, /obj/item/teleporter_kit))
+		return
+	
+	var/obj/item/teleporter_kit/gadget = I
+	if(src.linked_teleporter)
+		balloon_alert(user, "The teleporter is already linked with another!")
+		return
+	if(linked_teleporter == src)
+		balloon_alert(user, "You can't link the teleporter with itself!")
+		return
+	src.linked_teleporter = linked_teleporter
+	balloon_alert(user, "You link both teleporters to each others.")
+
+	src.set_linked_teleporter(gadget)
+	gadget.set_linked_teleporter(src)
+	return
```

Comment:
don't use src, it's either just the var for the telegetting attackby'd, or gadget.var

Reply:
Done